geom/type_sequence.go

```diff
@@ -34,6 +34,17 @@ func NewSequence(coordinates []float64, ctype CoordinatesType) Sequence {
 	return Sequence{ctype, coordinates}
 }
 
+// validate checks the X and Y values in the sequence for NaNs and infinities.
+func (s Sequence) validate() error {
+	n := s.Length()
+	for i := 0; i < n; i++ {
+		if err := s.GetXY(i).validate(); err != nil {
+			return wrap(err, "invalid XY at index %d", i)
```

Comment:
`wrap()` is implemented with:
```
return fmt.Errorf(format+": %v", append(args, err)...)
```

What is the reason for this approach? I've been under the impression that the idiomatic way of wrapping an error is to use the `%w` verb:
```
fmt.Errorf(format+": %w", append(args, err)...)
```

From https://go.dev/blog/go1.13-errors:

> Wrapping an error with %w makes it available to `errors.Is` and `errors.As`

Reply:
Wrapping errors non-opaquely (either with `%w` or with juju style wrappers etc) isn't too useful to users of the library, because we're not (yet?) exposing any of the concrete error types, such as `validationError`. So even if we used `%w` in the `wrap` implementation, users can't use `errors.Is` and `errors.As` anyway.

Whether using `%v` or `%w`, the actually error text will be the same.

I think you raise a good point though: it seems pretty idiomatic in Go to use `%w` for wrapping, even if it may not be too useful for end-users i.e. `%w` is a good default unless you explicitly want an opaque wrapping. We don't _need_ an opaque wrapping here, since the error types aren't exposed to users anyway. I'll raise an issue for this (might be a good first issue for someone new to the project to pick up).

Reply:
Ticket is here: https://github.com/peterstace/simplefeatures/issues/416